@here/harp-mapview/lib/Utils.ts

```diff
@@ -318,27 +313,12 @@ export namespace MapViewUtils {
      * you.
      */
     export function calculateZoomLevelFromHeight(height: number, mapView: MapView): number {
-        const targetSubdivisions = 2;
-        const biasedHeight = height * mapView.zoomLevelBias;
-        const visibleHeight =
-            2 * biasedHeight * Math.tan(geoUtils.MathUtils.degToRad(mapView.camera.fov / 2));
-
-        /**
-         * Use half the circumference as a reference point. This makes a few assumptions:
-         * 1. That the equitorial circumference is close enough to the vertical circumference to
-         *    not matter.
-         * 2. That distance is consistent throughout the globe. This isn't necessarily the case for
-         *    mercator projections, but shouldn't be a huge problem since the geometry is distorted
-         *    anyway.
-         * 3. That in most cases the curvature of the earth won't have a large impact. In the case
-         *    of zooming out to view the whole globe with a sphere projection, it will subdivide a
-         *    bit more than it would otherwise, but it shouldn't be much of an issue.
-         */
-        const halfCircumference = EarthConstants.EQUATORIAL_CIRCUMFERENCE / 2;
-        const targetSize = visibleHeight / targetSubdivisions;
-
-        const zoomLevel = Math.log2(halfCircumference / targetSize);
-        return geoUtils.MathUtils.clamp(zoomLevel, mapView.minZoomLevel, mapView.maxZoomLevel);
+        const tileSize = (256 * height) / mapView.focalLength;
```

Comment:
You will get different zoom levels for different size displays, I'm not sure this is what you want.